cmd/update/volume_move.go

```diff
@@ -0,0 +1,383 @@
+				return nil
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) getVolume(ctx context.Context, nsID id.Namespace) (vol *model.Volume, err error) {
+	useIDs, err := c.config.UseIDs()
+	if err != nil {
+		return nil, err
+	}
+
+	if useIDs {
+		vol, err = c.client.GetVolume(ctx, nsID, id.Volume(c.volumeID))
+		if err != nil {
+			return nil, err
+		}
+		return vol, nil
+	} else {
+		vol, err = c.client.GetVolumeByName(ctx, nsID, c.volumeID)
+		if err != nil {
+			return nil, err
+		}
+		return vol, nil
+	}
+}
+
+func newMoveCmd(w io.Writer, client interfaces.Client, config interfaces.ConfigProvider) *cobra.Command {
+	c := &volumeMoveCommand{
+		config: config,
+		client: client,
+		writer: w,
+	}
+
+	cobraCommand := &cobra.Command{
+		Use:   "move [volume name] [source node] [destination node]",
+		Short: "Updates a volume's placement",
+		Example: `
++$ storageos update volume move my-volume-name source-node-id destination-node-id --namespace my-namespace-name --timeout 10m
++$ storageos update volume move pvc-6d531e0e-fc5a-4b35-b752-bf75fc5dc690 38b7f402-c602-49f9-96b7-bc213d02a737 d4e910ac-394b-4fdb-be51-27822c15b682 --namespace c65e6116-ed59-4869-bf97-4fef5b42ac8e --timeout 30m
+`,
+		Args: argwrappers.WrapInvalidArgsError(func(cmd *cobra.Command, args []string) error {
+			if len(args) != 3 {
+				return clierr.NewErrInvalidArgNum(args, 3, "storageos update volume move [volume] [source node] [destination node]")
+			}
+
+			if args[0] == "" {
+				return errors.New("volume ID missing")
+			}
+			if args[1] == "" {
+				return errors.New("[source] node ID missing")
+			}
+			if args[2] == "" {
+				return errors.New("[destination] node ID missing")
+			}
+
+			c.volumeID = args[0]
+			c.fromNode = args[1]
+			c.toNode = args[2]
+
+			if c.fromNode == c.toNode {
+				return errors.New("[source] node ID cannot be the same as [destination] node ID")
+			}
+
+			return nil
+		}),
+
+		PreRunE: argwrappers.WrapInvalidArgsError(func(_ *cobra.Command, args []string) error {
+			c.display = SelectDisplayer(c.config)
+
+			ns, err := c.config.Namespace()
+			if err != nil {
+				return err
+			}
+
+			if ns == "" {
+				return clierr.ErrNoNamespaceSpecified
+			}
+			c.namespace = ns
+
+			return nil
+
+		}),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			run := runwrappers.Chain(
+				runwrappers.RunWithTimeout(c.config),
+				runwrappers.EnsureNamespaceSetWhenUseIDs(c.config),
+				runwrappers.AuthenticateClient(c.config, c.client),
+			)(c.runWithCtx)
+			return run(context.Background(), cmd, args)
+		},
+
+		SilenceUsage: true,
+	}
+
+	c.useCAS = flagutil.SupportCAS(cobraCommand.Flags(), &c.casVersion)
+	flagutil.SupportAsync(cobraCommand.Flags(), &c.useAsync)
+
+	// flags
+	cobraCommand.Flags().DurationVar(&c.timeout, "timeout", time.Minute*10, "timeout for this operation, directly related to how much data a volume has to sync. defaults to 10m")
```

Comment:
How does this interact with line 366 (`runwrappers.RunWithTimeout(c.config)`), does this flag set the config value?